src/compose/ComposeBox.js

```diff
@@ -198,7 +198,8 @@ export default class ComposeBox extends PureComponent<Props, State> {
     const { actions, draft, narrow } = this.props;
     const { message } = this.state;
 
-    if (message.trim().length === 0) {
+    if (message.trim().length === 0 && draft.length !== 0) {
```

Comment:
This will change the behavior, as we currently do remove the draft key from the store and not just ignore it. 

You can certainly improve the logic by adding a conditional `dispatch` in the `draftRemove` function.